core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -176,7 +170,7 @@ class Partition(val topic: String,
       // to maintain the decision maker controller's epoch in the zookeeper path
       controllerEpoch = partitionStateInfo.controllerEpoch
       // add replicas that are new
-      val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r)).toSet
+      val newInSyncReplicas = partitionStateInfo.isr.asScala.map(r => getOrCreateReplica(r, partitionStateInfo.isNew)).toSet
```

Comment:
After more thought, I think it simpler to just specify the isNew flag so that future developer doesn't need to think about it. I have updated the patch to do so.